app/controllers/application_controller.rb

```diff
@@ -1797,6 +1797,15 @@ def task_supported?(typ)
     end
 
     vms = VmOrTemplate.where(:id => vm_ids)
+    if "migrate" == typ
+      if vms.map(&:ext_management_system).uniq.compact.any? do |ems|
+        ems.respond_to?("supports_#{typ}_for_all?") && !ems.send("supports_#{typ}_for_all?", vms)
```

Comment:
Yes, please don't do that. No more magical send with string interpolation, and **especially** not when you *know* the type is `migrate` :).